src/Sylius/Component/Product/Resolver/DefaultProductVariantResolver.php

```diff
@@ -13,17 +13,24 @@
 
 namespace Sylius\Component\Product\Resolver;
 
+use Sylius\Component\Product\Repository\ProductVariantRepositoryInterface;
 use Sylius\Component\Product\Model\ProductInterface;
 use Sylius\Component\Product\Model\ProductVariantInterface;
 
 final class DefaultProductVariantResolver implements ProductVariantResolverInterface
 {
+    public function __construct(private ProductVariantRepositoryInterface $productVariantRepository)
```

Comment:
Due to backward compatibility, you should add this argument as optional and leave the old behaviour when this service is not present, and add a deprecation in 1.13 for not using it
```suggestion
    public function __construct(private ?ProductVariantRepositoryInterface $productVariantRepository = null)
```